### PR TITLE
[WIP] Private access services implementation

### PR DESCRIPTION
**Pull request by EvgenyAgafonchikov**
#### Subnet update
INPUT (under subnet.properties in PUT vnet and PUT subnet requests) 
```
            "privateAccessServices": [  
                {  
                    "service": “Microsoft.Storage” 
                }, 
                {  
                    "service": “Microsoft.Sql” 
                } 
            ], 
```

OUTPUT (in PUT/GET vnet, PUT/GET subnet responses): 
```
            "privateAccessServices": [  
                {  
                    "provisioningState": “Updating|Deleting|Failed|Succeeded”, 
                    "service": “Microsoft.Storage”, 
                    "locations": ["eastus", "westus"] 
                }, 
                {  
                    "provisioningState": “Updating|Deleting|Failed|Succeeded”, 
                    "service": “Microsoft.Sql”, 
                    "locations": ["eastus"] 
                } 
            ],
```
#### New feature
INPUT
```
GET /subscriptions/{subscriptionId}/providers/Microsoft.Network/locations/{location}/privateAccessServices?api-version=2017-06-01 
```
OUTPUT: 
```
{ 
  "value": [ 
    { 
      "id": "/subscriptions/s1/providers/Microsoft.Network/privateAccessServices/Microsoft.Storage", 
      "name": "Microsoft.Storage", 
      "type": "Microsoft.Network/privateAccessServices" 
    }, 
    { 
      "id": "/subscriptions/s1/providers/Microsoft.Network/privateAccessServices/Microsoft.Sql", 
      "name": "Microsoft.Sql", 
      "type": "Microsoft.Network/privateAccessServices" 
    } 
  ] 
}
```

This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process. 

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [x] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] The spec follows the guidelines described in the [Swagger checklist](../documentation/swagger-checklist.md) document.
  - [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

File: arm-network/2017-06-01/swagger/privateAccessServices.json [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+          "description": "List of available private access service values in a region."
+        },
+        "nextLink": {
+          "type": "string",
+          "description": "The URL to get the next set of results."
+        }
+      },
+      "description": "Response for the ListAvailablePrivateAccessServices API service call."
+    },
+    "PrivateAccessServiceResult": {
```
Comment (veronicagg) [inline arm-network/2017-06-01/swagger/privateAccessServices.json]: is this model an azure resource? should id, name and type be read only? (if it's an azure resource, you may want to tag it as x-ms-azure-resource.

**Review by veronicagg — CHANGES_REQUESTED** [review]
Is the privateAccessServices.json operation already deployed to ARM?
x-ms-examples are missing in the new spec for swagger model validation - https://travis-ci.org/Azure/azure-rest-api-specs/jobs/248463464#L1328 
will you be adding them? More info about this validation at https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist.
Adding @ravbhatnagar in the review, since you're adding a new file.

File: arm-network/2017-06-01/swagger/virtualNetwork.json [hunk 2]
```diff
@@ -801,6 +808,25 @@
         }
       }
     },
+    "PrivateAccessServicePropertiesFormat": {
```
Comment (EvgenyAgafonchikov) [inline arm-network/2017-06-01/swagger/virtualNetwork.json]: updated

File: arm-network/2017-06-01/swagger/virtualNetwork.json [hunk 3]
```diff
@@ -780,6 +780,13 @@
           "$ref": "./routeTable.json#/definitions/RouteTable",
           "description": "The reference of the RouteTable resource."
         },
+        "privateAccessServices": {
```
Comment (EvgenyAgafonchikov) [inline arm-network/2017-06-01/swagger/virtualNetwork.json]: Yes, it was. I've added spec extractions to clarify the changes, please find them in the top PR's comment.

**Comment by ravbhatnagar** [comment]
ACK - ARM sign off. This has been already reviewed and approved by ARM.

File: arm-network/2017-06-01/examples/SubnetCreate.json [hunk 4]
```diff
@@ -0,0 +1,32 @@
+{
+  "parameters": {
```
Comment (veronicagg) [inline arm-network/2017-06-01/examples/SubnetCreate.json]: parameters are not properly specified in the example, please add "subnetName" to the parameters and "subnetParameters" with its properties "name" and "addessPrefix" inside.

File: arm-network/2017-06-01/examples/SubnetPrivateAccessServiceCreate.json [hunk 5]
```diff
@@ -0,0 +1,49 @@
+{
+  "parameters": {
```
Comment (veronicagg) [inline arm-network/2017-06-01/examples/SubnetPrivateAccessServiceCreate.json]: similar comment as in the other example, missing required parameters, "subnetName", "subnetParamaters" with its properties "name", "addressPrefix" and "privateAccessServices" inside.

**Review by veronicagg — CHANGES_REQUESTED** [review]
couple more comments, to make review easier, please add commits on top of existing instead of removing old commits and commiting again. I'll squash for the final merge anyway. thanks.

File: arm-network/2017-06-01/examples/SubnetCreate.json (hunk 4 above)
Comment (EvgenyAgafonchikov) [inline arm-network/2017-06-01/examples/SubnetCreate.json]: Updated, please have a look. I think there should be no "name" property under "subnetParameters" as it is placed one level above

File: arm-network/2017-06-01/swagger/privateAccessServices.json (hunk 1 above)
Comment (EvgenyAgafonchikov) [inline arm-network/2017-06-01/swagger/privateAccessServices.json]: updated as inherited from SubResource

**Comment by DeepakRajendranMsft** [comment]
@veronicagg could you please take a look and merge if all comments have been resolved?

File: arm-network/2017-06-01/swagger/privateAccessServices.json (hunk 1 above)
Comment (veronicagg) [inline arm-network/2017-06-01/swagger/privateAccessServices.json]: ok, thanks, is here a reason for name and type to not be read-only? https://travis-ci.org/Azure/azure-rest-api-specs/jobs/249523709#L1307

**Review by veronicagg — CHANGES_REQUESTED** [review]
thanks for addressing the comments. One more question, other than that looks good.

**Comment by DeepakRajendranMsft** [comment]
@veronicagg I am working on a PR that adds read only properties for all SDN resources (nics, vnet, nsg, routes, lb) i will add them as part of that PR. Currently, none of the SDN resources have readonly flags.

**Comment by veronicagg** [comment]
@DeepakRajendranMsft ok, thanks.

**Review by veronicagg — APPROVED** [review]

**Pull request merged**